src/Servers/HttpSys/test/FunctionalTests/Listener/RequestHeaderTests.cs

```diff
@@ -15,10 +15,37 @@ namespace Microsoft.AspNetCore.Server.HttpSys.Listener;
 
 public class RequestHeaderTests
 {
+    [ConditionalFact]
+    public async Task RequestHeaders_RemoveHeaders_Success()
+    {
+        string address;
+        using (var server = Utilities.CreateHttpServer(out address))
+        {
+            string[] customValues = new string[] { "custom1, and custom测试2", "custom3" };
+            Task responseTask = SendRequestAsync(address, "Custom-Header", customValues, Encoding.UTF8);
```

Comment:
Does this send a Connection header? Make sure it sends a "Connection: close" header.